google-cloud-clients/google-cloud-datastore/src/main/java/com/google/cloud/datastore/QueryResultsImpl.java

```diff
@@ -124,4 +125,19 @@ public int getSkippedResults() {
   public MoreResultsType getMoreResults() {
     return moreResults;
   }
+
+  @Override
+  public int countEntities() {
+    int count = 0;
+    Integer limit = ((KeyQuery) this.query).getLimit();
+    if (null != limit && limit >= 0) {
+      QueryResultBatch queryResultBatch = runQueryResponsePb.getBatch();
+      count = queryResultBatch.getSkippedResults();
+      while (queryResultBatch.getMoreResults()
```

Comment:
I have a doubt for calling `queryResultBatch.getMoreResults()`... Does it lead to blocking call?